sql/src/main/codegen/includes/from.ftl

```diff
@@ -0,0 +1,355 @@
+                     joinType,
+                     e2,
+                     JoinConditionType.NONE.symbol(SqlParserPos.ZERO),
+                     null);
+             }
+         |
+             <CROSS> { joinType = JoinType.CROSS.symbol(getPos()); } <APPLY>
+             e2 = DruidTableRef2(true) {
+                 if (!this.conformance.isApplyAllowed()) {
+                     throw SqlUtil.newContextException(getPos(), RESOURCE.applyNotAllowed());
+                 }
+                 e = new SqlJoin(joinType.getParserPosition(),
+                     e,
+                     SqlLiteral.createBoolean(false, joinType.getParserPosition()),
+                     joinType,
+                     e2,
+                     JoinConditionType.NONE.symbol(SqlParserPos.ZERO),
+                     null);
+             }
+         |
+             <OUTER> { joinType = JoinType.LEFT.symbol(getPos()); } <APPLY>
+             e2 = DruidTableRef2(true) {
+                 if (!this.conformance.isApplyAllowed()) {
+                     throw SqlUtil.newContextException(getPos(), RESOURCE.applyNotAllowed());
+                 }
+                 e = new SqlJoin(joinType.getParserPosition(),
+                     e,
+                     SqlLiteral.createBoolean(false, joinType.getParserPosition()),
+                     joinType,
+                     e2,
+                     JoinConditionType.ON.symbol(SqlParserPos.ZERO),
+                     SqlLiteral.createBoolean(true, joinType.getParserPosition()));
+             }
+         )
+     )*
+     {
+         return e;
+     }
+ }
+
+ /**
+  * Parses a table reference in a FROM clause, not lateral unless LATERAL
+  * is explicitly specified.
+  */
+ SqlNode DruidTableRef() :
+ {
+     final SqlNode e;
+ }
+ {
+     e = DruidTableRef2(false) { return e; }
+ }
+
+ /**
+  * Parses a table reference in a FROM clause.
+  */
+ SqlNode DruidTableRef2(boolean lateral) :
+ {
+     List<SqlNode> paramList;
+     SqlNode tableRef;
+     final SqlNode over;
+     final SqlNode snapshot;
+     final SqlNode match;
+     SqlNodeList extendList = null;
+     final SqlIdentifier alias;
+     final Span s, s2;
+     SqlNodeList args;
+     SqlNode sample;
+     boolean isBernoulli;
+     SqlNumericLiteral samplePercentage;
+     boolean isRepeatable = false;
+     int repeatableSeed = 0;
+     SqlNodeList columnAliasList = null;
+     SqlUnnestOperator unnestOp = SqlStdOperatorTable.UNNEST;
+ }
+ {
+     (
+         LOOKAHEAD(2)
+         tableRef = CompoundIdentifier()
+         [
+             paramList = FunctionParameterList(ExprContext.ACCEPT_NONCURSOR)
+             {
+                 tableRef = ParameterizeOperator.PARAM.createCall(tableRef, paramList);
+             }
+         ]
+         [
+             [ <EXTEND> ]
+             extendList = ExtendList() {
+                 tableRef = extend(tableRef, extendList);
+             }
+         ]
+         over = TableOverOpt() {
```

Comment:
Remove the following lines:
```text
         [
             [ <EXTEND> ]
             extendList = ExtendList() {
                 tableRef = extend(tableRef, extendList);
             }
         ]
```

Druid supports `EXTEND` now, but not in the same place as Phoenix. Removing these lines avoids the need for us to parse, then later reject, the Phoenix syntax.

Reply:
Thanks for catching this! Fixed